infor-design/enterprise
# 1038 - Configure Node-Sass to only import Sass partials once

**Explain the _details_ for making this change. What existing problem does the pull request solve?**
This PR makes a change to the build system that configures `node-sass` to only import required partials once, instead of allowing them to be imported (and therefore generate rules) multiple times.  This is done by incorporating the [`node-sass-once-importer`](https://github.com/maoberlehner/node-sass-magic-importer/tree/master/packages/node-sass-once-importer) into the build system.

What the importer allows is for imports of SASS dependencies in any files that need them (for example, `_button.scss` to be included in Pager and Listview, and to not be imported more than once), which is similar to how the ES6 importing system currently works.

This change will allow us to liberally place dependencies on other SASS partials inside of mid-to-complex components, and expect the source code to be cleaner in the process.  This will also improve our ability to split the SASS source code into many smaller files and correctly pull in only what's necessary when generating custom builds.

See the changes at the top of `_listview.scss` in this PR for an example of how it works.

**Related github/jira issue (required)**:
Closes #1038 

**Steps necessary to review your pull request (required)**:
_Test that the build actually runs:_
- Pull this branch
- Run `npm prune && npm i` to include the new dependency
- Run a build/minify/verify with `npm run clean && npm run build && npm run minify && npm run build:verify`
- Ensure this process completes with no errors and the files are verified properly.

_To test that there are no duplicate imports:_
- Start the demoapp and navigate to http://localhost:4000/components/listview/test-empty-message.html
- Open a dev tools instance.
- Find the `div.empty-message` element in the Inspector tab.  Ensure that the CSS rules coming in from `_emptymessage.scss` do not show up more than once in the Styles tab (which would indicate multiple imports):
![Screen Shot 2019-08-19 at 12 39 07 PM](https://user-images.githubusercontent.com/3249601/63283446-38bc2400-c27f-11e9-9e08-53185bcadd3b.png)
- Open http://localhost:4000/components/listview/example-paging.html
- In dev tools, inspect the `li.pager-prev > a` element in the Pager Toolbar.
- Ensure that none of the CSS rules coming from `_pager.scss` are duplicated:
![Screen Shot 2019-08-19 at 12 44 53 PM](https://user-images.githubusercontent.com/3249601/63283552-8c2e7200-c27f-11e9-89ea-abbc82b1aa34.png)

**NOTE:** Listview is the only component modified for the new import system in this PR.  All other components should be the same, so it's worth a quick smoke test of any other components you like to ensure nothing else has changed unexpectedly.

**Included in this Pull Request**:
